README.md

```diff
@@ -0,0 +1,223 @@
+| ***source***              | The source to use                                                                      | **authzv2** / **meauthz**                      |
+| ***sources***             | Object with possible authz sources and their configurations                           | ```{ authzv2: { _config_ }}```    |
+| authzv2: ***url***        | Url of the authz api (v2) You can find this on the api-store                          | *\_URL\_OAUTHZ\_*                 |
+| authzv2: ***apikey***     | Api key. You will need to create an application with a contract with the authz api    | *\_APIKEY\_*                      |
+
+##### Example:
+
+```javascript
+const { config } = require('@digipolis/authz');
+
+config({
+  debug: true,
+  source: 'authzv2',
+  sources: {
+    authzv2: {
+      url:  '_URL_AUTHZ_',
+      apikey: '_APIKEY_',
+      applicationId: '_APPLICATION_ID_',
+    },
+    meauthz: {
+      url:  '_URL_AUTHZ_',
+      apikey: '_APIKEY_',
+      applicationId: '_APPLICATION_ID_',
+    },
+  },
+});
+```
+## Usage
+
+The module can be used as an express middleware or as a function. The parameter is a string to check a single permission or an array to check multiple permissions. If a permission is missing an Error of the type [**PermissionError**](#permissionError) will be thrown.
+
+An example of this can be found in the documentation below and in the example folder.
+
+#### Usage as express middleware:
+Configuration should be done before usage.
+
+```javascript
+const { Router } = require('express');
+const { hasPermission } = require('@digipolis/authz');
+
+const router = new Router();
+
+// Check single permission in default source
+router.get('/', hasPermission('login-app'), controller);
+// Check mutiple permissions in default source
+router.get('/', hasPermission(['login-app', 'admin-app']), controller);
+// Check permission in default meauthz source
+router.get('/', hasPermission('login-app', 'meauthz'), controller);
+
+```
+#### Usage as function:
+Configuration should be done before usage.
+
+```javascript
+const { checkPermission } = require('@digipolis/authz');
+const { create } = require('./itemcreator.service');
+
+async function createSomething(params, usertoken) {
+    await checkPermission(usertoken, 'login-app'); //throws error if invalid
+    await checkPermission(usertoken, ['login-app', 'use-app']); //throws error if invalid
+    await checkPermission(usertoken, 'login-app', 'meauthz'); //throws error if invalid
+    return create(params);
+}
+```
+
+#### Permissionerror:
+
+##### Model
+
+```javascript
+{
+  ...extends_default_javascript_error
+  name: 'PermissionError',
+  message: 'Failed to retrieve permissions.' // example
+  detail: {
+    message: 'Invalid Token' // example
+  }
+}
+```
+##### Error messages:
+
+-  `ApplicationId not configured.`
+-  `Authzv2 not configured.`
+-  `Trying to set configuration more then once.`
+-  `Force set configuration.`
```

Comment:
"Force set configuration" & "Trying to set configuration more then once" komt precies niet voor als error, maar enkel in logmessages?

Reply:
correct, verwijderd